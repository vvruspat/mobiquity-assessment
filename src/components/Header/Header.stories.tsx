import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Header } from "./Header";
import "../../index.css";

export default {
  title: "Components/Header",
  component: Header,
} as ComponentMeta<typeof Header>;

const Template: ComponentStory<typeof Header> = (args) => (
  <Header {...args}></Header>
);

export const HeaderTemplate = Template.bind({});

HeaderTemplate.args = {};
