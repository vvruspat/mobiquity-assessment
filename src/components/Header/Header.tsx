import {
  DetailedHTMLProps,
  FC,
  HTMLAttributes,
  PropsWithChildren,
  useCallback,
  useEffect,
  useMemo,
} from "react";
import cn from "clsx";

import { Menu, MenuItemOption } from "../Menu";
import { useSeasonsStore } from "../../stores/seasons";
import { fetchSeasons } from "../../api/Api";

import { ReactComponent as LogoIcon } from "./assets/logo.svg";

import s from "./Header.module.css";

type HeaderProps = DetailedHTMLProps<
  HTMLAttributes<HTMLDivElement>,
  HTMLDivElement
>;

/**
 * Page header component
 */
export const Header: FC<PropsWithChildren<HeaderProps>> = ({
  className,
  ...htmlProps
}) => {
  const { seasons, setSeasons, setSelectedSeason } = useSeasonsStore(
    (state) => state
  );

  /** Load list of available seasons */
  useEffect(() => {
    (async () => {
      try {
        const seasonsResponse = await fetchSeasons();

        setSeasons(
          seasonsResponse.MRData.SeasonTable.Seasons
            /** Remove current year, because we dont have winner yet */
            .slice(0, -1)
        );
      } catch (e) {
        setSeasons([]);
      }
    })();
  }, [setSeasons]);

  /** Map seasons to menu items */
  const seasonsOptions: MenuItemOption[] = useMemo(
    () =>
      seasons
        .map((season) => ({ key: season.season, value: season.season }))
        .reverse(),
    [seasons]
  );

  const onSeasonSelect = useCallback(
    (selected: MenuItemOption) => {
      setSelectedSeason(selected.key);
    },
    [setSelectedSeason]
  );

  return (
    <header className={cn(s.header, className)} {...htmlProps}>
      <div className={s.header_content}>
        <LogoIcon />
      </div>
      <Menu options={seasonsOptions} onSelect={onSeasonSelect} />
    </header>
  );
};
