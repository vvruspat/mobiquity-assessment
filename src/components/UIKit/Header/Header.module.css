.header {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    padding: var(--padding-large) 0;
    padding-top: 24px;
    box-sizing: border-box;
    height: 124px;
    background: linear-gradient(81.12deg, #CE2D1E 3.34%, #4A1E1E 98.84%);
    border-bottom-left-radius: var(--padding-large);
    border-bottom-right-radius: var(--padding-large);
}

.header_content {
    display: flex;
    justify-content: space-between;
    padding: 0 var(--padding-large);
}