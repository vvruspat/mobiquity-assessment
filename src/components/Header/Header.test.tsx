import { render, screen } from "@testing-library/react";
import { Header } from "./Header";

test("renders header", () => {
  render(<Header />);

  const el = screen.getByTestId("header");
  expect(el).toMatchSnapshot();

  const menu = screen.getByTestId("menu");
  expect(menu).toBeInTheDocument();
});
