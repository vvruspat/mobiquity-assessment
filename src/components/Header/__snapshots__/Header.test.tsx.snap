// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header 1`] = `
<header
  class="header"
  data-testid="header"
>
  <div
    class="header_content"
  >
    <svg>
      logo.svg
    </svg>
    <button
      class="button icon-button theme_button"
      data-testid="button"
    >
      <svg>
        dark.svg
      </svg>
    </button>
  </div>
  <div
    class="menu"
    data-testid="menu"
    dir="ltr"
  />
</header>
`;
